app/main/controllers/api/RTMediaJsonApi.php

```diff
@@ -1065,15 +1065,23 @@ function rtmedia_api_process_rtmedia_gallery_request() {
 		$media_type_temp  = filter_input( INPUT_POST, 'media_type', FILTER_SANITIZE_STRING );
 		$media_type_array = filter_input( INPUT_POST, 'media_type', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY );
 
-		if ( ! empty( $media_type_temp ) || ! empty( $media_type_array ) ) {
-			if ( ! is_array( $media_type_array ) ) {
-				$media_type = explode( ',', $media_type_array );
-			} else {
-				$media_type = $media_type_temp;
-			}
-			//Check array for currently allowed media types
-			$media_type = array_intersect( $media_type, $allowed_types );
-		}
+		if ( ! empty( $media_type_temp ) ) {
+			/**
+			 * Explode by , when string found
+			 */
+			$media_type = explode( ',', $media_type_temp );
+		} elseif ( ! empty( $media_type_array ) && is_array( $media_type_array ) ) {
+			/**
+			 * Set media_type to passed array
+			 */
+			$media_type = $media_type_array;
+		}
+		/**
+		 * Check array for currently allowed media types
+		 */
+		$media_type = array_intersect( $media_type, $allowed_types );
```

Comment:
If both `if else if` condition not executed then `$media_type` will throw a notice. no?

Reply:
Default `$media_type` is defined before, so if both is not executed, the default will be processed.